client/components/track-clicks/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import storeShape from 'react-redux/lib/utils/storeShape';
+import { cloneElement, Children, Component, PropTypes } from 'react';
+import { constant, isFunction, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import pick from 'lib/pick';
+import { GUIDED_TOUR_TRACKED_CLICK } from 'state/action-types';
+
+const stringShape = PropTypes.oneOfType( [
+	PropTypes.string,
+	PropTypes.arrayOf( PropTypes.string )
+] );
+
+export default class TrackClicks extends Component {
+	static propTypes = {
+		children: PropTypes.element.isRequired,
+		mapPropsToAction: PropTypes.func,
+		fields: stringShape,
+	}
+
+	static defaultProps = {
+		mapPropsToAction: constant( {} ),
+	}
+
+	static contextTypes = { store: storeShape };
+
+	constructor( props, context ) {
+		super( props, context );
+		const store = ( props.store || context.store );
+		this.dispatch = store ? store.dispatch : noop;
+	}
+
+	track = ( child ) => {
+		const action = this.mapPropsToAction( child.props );
+
+		this.dispatch( {
+			type: GUIDED_TOUR_TRACKED_CLICK,
+			component: child.type.name,
+			...action,
+		} );
+	}
+
+	mapPropsToAction( props ) {
+		const { fields, mapPropsToAction } = this.props;
+		return fields
+			? pick( props, fields )
+			: mapPropsToAction( props );
+	}
+
+	render() {
+		const child = Children.only( this.props.children );
+		const props = {
+			...child.props,
+			onClick: ( event ) => {
```

Comment:
> Does this also cover taps on mobile devices?

I haven't tested, but I expect so, since we're interacting with React and its SyntheticEvent, so I assume that synthetic _clics_ are mapped to the appropriate event for each device. But **@todo test this**.

> And: could we make this more generic by maybe defaulting to onClick but providing an override to e.g. also use it for catching onHover events?

Yeah, I think that'll be useful. With ES6's dynamic object properties, this is convenient enough:

```js
const { eventType = 'onClick' } = this.props;
const props = {
  ...child.props,
  [ eventType ]: ( event ) => {
    if ( isFunction( child.props[ eventType ] ) ) {
      child.props[ eventType ].call( child, event );
    }
  …
```